submissions/unabyband/oop_exercise/index.js

```diff
@@ -0,0 +1,103 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/unabyband/a-tiny-JS-world
+   Web app: https://unabyband.github.io/a-tiny-JS-world/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Inhabitant {
+  constructor(species, name, gender, legs, saying, friends) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.legs = legs;
+    this.saying = saying;
+    this.friends = friends;
+  }
+
+  printInhabitant() {
+    const friendsToPrint = this.friends
+      ? this.friends.map((friend) => friend.name).join(", ")
+      : `${this.species} has no friends!`;
+    
+    return [this.species, 
+      `<strong>${this.name}</strong>`, 
+      this.gender, 
+      `On ${this.legs} legs`, 
+      `<em>${this.saying}</em>`, 
+      friendsToPrint].join('; ');
+  }  
+}
+
+class Human extends Inhabitant {
+  constructor(name, gender, saying, friends) {
+    const legs = 2;
+    const hands = 2;
+    super("human", name, gender, legs, saying, friends);
+    this.hands = hands;
+  }
+
+  printInhabitant() {
+    return [this.species,
+        `<strong>${this.name}</strong>`,
+        this.gender,
+        `On ${this.legs} legs`,
+        `Has ${this.hands} hands`,
+        `<em>${this.saying}</em>`,
+        this.friends.map((friend) => friend.name).join(", ")].join('; ');
```

Comment:
Now re-use parent's version of the method and add whatever this class needs to add.
Let's have hands at the end for the sake of simplicity and in favour of OOP compliance.

Alternative would be to keep properties as array all way down until you need to print them. This way you could insert hands somewhere after legs or elsewhere. But this too much change. Please just keep this in mind. And attach hands at the end of the parent's method output.

Reply:
I've made it, please review. And sorry for delayed answer, because of missile attacks we don't have an electricity and cable internet.
Thank you